samcli/commands/init/init_templates.py

```diff
@@ -0,0 +1,150 @@
+        self._repo_url = "https://github.com/awslabs/aws-sam-cli-app-templates.git"
+        self._repo_name = "aws-sam-cli-app-templates"
+        self.repo_path = None
+        self.clone_attempted = False
+        self._no_interactive = no_interactive
+        self._auto_clone = auto_clone
+
+    def prompt_for_location(self, runtime, dependency_manager):
+        options = self.init_options(runtime, dependency_manager)
+        choices = map(str, range(1, len(options) + 1))
+        choice_num = 1
+        for o in options:
+            if o.get("displayName") is not None:
+                msg = str(choice_num) + " - " + o.get("displayName")
+                click.echo(msg)
+            else:
+                msg = (
+                    str(choice_num)
+                    + " - Default Template for runtime "
+                    + runtime
+                    + " with dependency manager "
+                    + dependency_manager
+                )
+                click.echo(msg)
+            choice_num = choice_num + 1
+        choice = click.prompt("Template Selection", type=click.Choice(choices), show_choices=False)
+        template_md = options[int(choice) - 1]  # zero index
+        if template_md.get("init_location") is not None:
+            return template_md["init_location"]
+        elif template_md.get("directory") is not None:
+            return os.path.join(self.repo_path, template_md["directory"])
+        else:
+            raise UserException("Invalid template. This should not be possible, please raise an issue.")
+
+    def location_from_app_template(self, runtime, dependency_manager, app_template):
+        options = self.init_options(runtime, dependency_manager)
+        try:
+            template = next(item for item in options if self._check_app_template(item, app_template))
+            if template.get("init_location") is not None:
+                return template["init_location"]
+            elif template.get("directory") is not None:
+                return os.path.join(self.repo_path, template["directory"])
+            else:
+                raise UserException("Invalid template. This should not be possible, please raise an issue.")
+        except StopIteration:
+            msg = "Can't find application template " + app_template + " - check valid values in interactive init."
+            raise UserException(msg)
+
+    def _check_app_template(self, entry, app_template):
+        return entry["appTemplate"] == app_template
+
+    def init_options(self, runtime, dependency_manager):
+        if self.clone_attempted is False:
+            self._clone_repo()
+        if self.repo_path is None:
+            return self._init_options_from_bundle(runtime, dependency_manager)
+        return self._init_options_from_manifest(runtime, dependency_manager)
+
+    def _init_options_from_manifest(self, runtime, dependency_manager):
+        manifest_path = os.path.join(self.repo_path, "manifest.json")
```

Comment:
can we set manifest file as `manifest.json`, and reference that as self.manifest_file or self.manifest?